src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+  }
+
+  applyDamage(healthPoints) {
+    this.health -= healthPoints;
+
+    if (this.health < 0) {
```

Comment:
The conditional check should be `this.health <= 0` to align with the task description that states an animal dies when its health becomes less than or equal to 0. Currently, the check is for `< 0`, which does not account for the case when health is exactly 0.